runtime/jcl/common/system.c

```diff
@@ -70,12 +70,46 @@ jstring JNICALL Java_java_lang_System_getEncoding(JNIEnv *env, jclass clazz, jin
 	return getEncoding(env, encodingType);
 }
 
+#if JAVA_SPEC_VERSION >= 18
+jstring JNICALL
+Java_java_lang_System_getSysPropBeforePropertiesInitialized(JNIEnv *env, jclass clazz, jint sysPropID)
```

Comment:
I'll merge this now, but in the future we could add the encodings to this function and then delete Java_java_lang_System_getEncoding. It does basically the same thing, just the name is inappropriate.

Reply:
Sure, I saw that native, thought it is dedicated for `PlatformEncoding`/`OSEncoding`/`FileEncoding`, and opted to create one for generic system properties.